sfm_pc/views.py

```diff
@@ -264,38 +267,36 @@ def download_zip(request):
     return resp
 
 
-class DownloadData(FormView):
+class DownloadData(TemplateView):
     template_name = 'download.html'
-    form_class = DownloadForm
-    success_url = reverse_lazy('download')
 
-    def form_valid(self, form):
+    def get_context_data(self):
+        context = super().get_context_data()
 
-        download_type = form.cleaned_data['download_type']
-        division_id = form.cleaned_data['division_id']
-        confidences = form.cleaned_data['confidences']
+        download_url = self.get_presigned_url()
 
-        download_class = download_classes[download_type]
-        iso = division_id[-2:]
-        filename = '{}_{}_{}.csv'.format(
-            download_type,
-            iso.upper(),
-            timezone.now().date().isoformat()
-        )
+        if download_url:
+            context['download_url'] = download_url
 
-        # Don't filter sources by division_id
-        if download_type == 'sources':
-            return download_class.render_to_csv_response(
-                filename,
-                confidences=confidences
-            )
-        else:
-            return download_class.render_to_csv_response(
-                filename,
-                division_id=division_id,
-                confidences=confidences
+        return context
+
+    def get_presigned_url(self):
+        s3_client = boto3.client('s3')
+        try:
+            response = s3_client.generate_presigned_url(
+                'get_object',
+                Params={
+                    'Bucket': 'sfm-public-data',
+                    'Key': 'sfm_download.zip'
+                },
+                ExpiresIn=3600
```

Comment:
Do you reckon this is long enough?

Reply:
Also I'm getting access denied from my local instance.

Reply:
Good question about long enough


And re access denied, did you follow the new instructions I added in the readme?
> #### Zip download config
> If you need to test the sfm_download.zip fuctionality, `cp .env.example .env` and add your AWS access tokens. Then run  `docker-compose run --rm app make download_to_s3` so you can upload the zip archive to S3. Once this is done, the "download" link on the `~/download` page should work.

The AWS access tokens need to be tokens for an IAM user that have access to the S3 bucket — I can add more information about that in the readme so it's more clear.

Reply:
Yes, please specify that the configured AWS user needs access to the bucket. I should have caught on, but the explicit warning helps. 😅

Reply:
No problem, explicitness is key for docs! I added that instruction at the last minute and should've refined. Thanks for inadvertently pointing out that weakness.

Reply:
Re: expiration, this expires in one hour: https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/s3.html#S3.Client.generate_presigned_url

That seems like plenty of time, what do you think?